src/controller/AutoCommissioner.cpp

```diff
@@ -502,9 +502,41 @@ CHIP_ERROR AutoCommissioner::CommissioningStepFinished(CHIP_ERROR err, Commissio
             SetDAC(report.Get<RequestedCertificate>().certificate);
             break;
         case CommissioningStage::kSendAttestationRequest:
-            // These don't need to be deep copied to local memory because they are used in this one step then never again.
-            mParams.SetAttestationElements(report.Get<AttestationResponse>().attestationElements)
-                .SetAttestationSignature(report.Get<AttestationResponse>().signature);
+            if (report.Get<AttestationResponse>().attestationElements.size() > kAttestationElementsLength)
+            {
+                ChipLogError(Controller, "AutoCommissioner attestationElements buffer size %d larger than cache size %d",
+                             static_cast<int>(report.Get<AttestationResponse>().attestationElements.size()),
+                             static_cast<int>(kAttestationElementsLength));
+                mParams.SetAttestationElements(report.Get<AttestationResponse>().attestationElements);
```

Comment:
But in this case we keep using the ByteSpan that points to some temporary memory... how is a consumer expected to know when it's safe to use the span and when it's not?

Seems like we should just error out when the size is not what we expected.....  And something somewhere needs to document how long it's ok to keep using the spans in mParams, though I guess "until the next commissioning starts" might be clear enough.

Reply:
I created GetAttestationX() methods to the AutoCommissioner which will always return valid ByteSpans, and these are now called from Android rather than going through the CommissioningParameters object. The CommissioningParameters GetAttestationX() methods are not safe to use after the given step completes. I added a warning to that header file:

>Warning: data is not deep copied to local memory and can only be read during PerformCommissioningStep.